src/unicorn_baseline/language/main.py

```diff
@@ -12,6 +12,200 @@
+                "hyperplastic_polyps",
+                "low_grade_dysplasia",
+                "non_informative",
+                "serrated_polyps",
+            ]
+            example[prediction_name] = [
+                1.0 if example.pop(key) in ["True", True] else 0.0 for key in keys
+            ]
+    elif task_id == 17:
+        for example in data:
+            example[prediction_name] = [
+                float(example.pop("lesion_1")),
+                float(example.pop("lesion_2")),
+                float(example.pop("lesion_3")),
+                float(example.pop("lesion_4")),
+                float(example.pop("lesion_5")),
+            ]
+    data = drop_keys_except(data, ["uid", prediction_name])
+    return data
+
 
 def run_language():
-    pass
+    # Read the task configuration, few-shots and test data
+    # We'll leverage the DRAGON baseline algorithm for this
+    algorithm = DragonBaseline()
+    algorithm.load()
+    algorithm.analyze()  # needed for verification of predictions
+    task_config = algorithm.task
+    few_shots = algorithm.df_train
+    test_data = algorithm.df_test
+    basepath = Path("/opt/app/unicorn_baseline/src/unicorn_baseline/language/")
+    setup_folder_structure(basepath, test_data)
+    print(f"Task description: {task_config}")
+
+    task_name = task_config.task_name
+    task_id = task_config.jobid
+
+    # Task specific preprocessing
+    if "Task16_" in task_name:
+        test_data["text"] = test_data["text_parts"].apply(task16_preprocessing)
+
+    # Perform data extraction
+    extractinate(
+        task_id=task_id,
+        model_name="phi4",
+        num_examples=0,
+        temperature=0.0,
+        max_context_len="max",
+        num_predict=512,
+        translate=False,
+        data_dir=basepath / "data",
+        output_dir=basepath / "output",
+        task_dir=basepath / "tasks",
+        n_runs=1,
+        verbose=False,
+        run_name="run",
+        reasoning_model=False,
+        seed=42,
+    )
+
+    # Wait for the predictions to be generated and saved
+    runpath = basepath / "output" / "run"
+    datafolder = wait_for_predictions(
+        runpath=runpath,
+        task_name=str(task_id),
+        timeout=300,
+        interval=10,
+    )
+
+    # Load the predictions
+    datapath = datafolder / "nlp-predictions-dataset.json"
+    with open(datapath, "r") as file:
+        predictions = json.load(file)
+
+    predictions = post_process_predictions(predictions, task_config)
+
+    # Save the predictions
+    test_predictions_path = OUTPUT_PATH / "nlp-predictions-dataset.json"
+    write_json_file(
+        location=test_predictions_path,
+        content=predictions,
+    )
+
+    # Verify the predictions
+    algorithm.test_predictions_path = test_predictions_path
+    algorithm.verify_predictions()
+
+    print(f"Saved neural representation to {test_predictions_path}")
+    return 0
+
+
+def write_json_file(*, location, content):
```

Comment:
Please use the general JSON save function, which includes sanitizing the content: https://github.com/DIAGNijmegen/unicorn_baseline/blob/cbef9f8dbb3f94e7c29ec09e1504519a71fbe91e/src/unicorn_baseline/io.py#L64